benchmark.c

```diff
@@ -69,10 +69,17 @@ int main(int argc, const char * const *argv)
 
 	printf("Will use %.2f KiB RAM\n", 0.125 * params.N * params.r);
 
+#ifdef _WIN32
```

Comment:
Here and in the other place, please:
1. Avoid source code duplication - move the `__declspec(thread)` and `__thread` to be before `static` (and on their own lines), so just those qualifiers are conditional.
2. Check for `_MSC_VER` (like we do in https://github.com/openwall/passwdqc), not `_WIN32`.

Reply:
> `__thread` to be before `static`

Oops, no, gcc doesn't like that. So have to do:

```diff
+++ b/benchmark.c
@@ -69,7 +69,13 @@ int main(int argc, const char * const *argv)
 
        printf("Will use %.2f KiB RAM\n", 0.125 * params.N * params.r);
 
-       static __thread union {
+       static
+#ifdef _MSC_VER
+       __declspec(thread)
+#else
+       __thread
+#endif
+       union {
                uint8_t u8[80];
                uint32_t u32[20];
        } src;
```

and similar in the other place.

Reply:
> and similar in the other place.

or actually not, since we have two instances in there. Let's then keep your other change almost as-is, only changing `_WIN32` to `_MSC_VER`.

Ideally, we'd define a macro in some header file, but there isn't an existing suitable yespower-internal header file for things like that. So let's not do that yet.